### DISCUSSION TITLE
[Feature Request] Some Components Feature Suggestions

### DISCUSSION THREAD

**Discussion by meimosor** [heroui-inc/heroui#4683] [Feature requests]
### Is your feature request related to a problem? Please describe.

Our project is preparing to migrate to nextUI, but these components are not included here. Therefore, I have compiled some common components, hoping to extend them to the nextUI component library for more people to choose and use. Here are some feature suggestions and screenshots:

### Describe the solution you'd like

1. DatePicker and TimePicker. such as:
![image](https://github.com/nextui-org/nextui/assets/31975932/30427945-99d0-4c2b-84da-ff42325ea540)
Reference link: [https://ant.design/components/date-picker](url)
2. Mentions,When you need to mention someone or something. such as:
![image](https://github.com/nextui-org/nextui/assets/31975932/514c1192-ebe8-4cdd-adcc-08d8406f1319)
Reference link: [https://ant.design/components/mentions](url)
3. Transfer,It is a select control essentially which can be use for selecting multiple items. such as:
![image](https://github.com/nextui-org/nextui/assets/31975932/8ab9b1ea-5ae5-4ccf-8f82-b8dfc80f3b02)
Reference link: [https://ant.design/components/transfer](url)
4. Layout, Expand layout usage scenarios. such as:
![image](https://github.com/nextui-org/nextui/assets/31975932/9a747269-f2cf-4b7f-bb75-8363b7a1d143)
Reference link: [https://ant.design/components/layout](url)
5. Timeline, such as:
![image](https://github.com/nextui-org/nextui/assets/31975932/454d61ef-7d17-4003-b6c5-c9dfd3f49d7d)
6. Drawer, such as:
![image](https://github.com/nextui-org/nextui/assets/31975932/7c5e2b08-919c-4190-bbe3-1d5778cd4a6a)
7. Tree, such as:
![image](https://github.com/nextui-org/nextui/assets/31975932/5f42c197-aacf-48f1-b63e-5b19006658ea)
8. Upload, Image or File Upload feats. such as:
![image](https://github.com/nextui-org/nextui/assets/31975932/30b84ce8-dea0-4675-8a7f-8e477cb1f4ac)

### Describe alternatives you've considered

I'm sorry for raising so many component functional requirements, but I was mainly conquered by the beauty of nextUI, so I hope it becomes more perfect and convenient. Finally, I suggest that the nextUI team add a roadmap so that everyone can see the development path of nextUI. Thank you to the nextUI team!

**Comment by ssijak** [comment]
I don't think some of these components should belong in the core library. Like timeline, for example

**Comment by Emestry** [comment]
toasts should be implemented

**Comment by siddiquiaffan** [comment]
Awesome! 

I think drawer should be implemented. Reference https://ui.shadcn.com/docs/components/drawer

ShadCn ui is also a great component library based on radix-ui

**Comment by 1111mp** [comment]
Hey guys, is there any documentation where I can get specifications for developing components? I currently have time to contribute something.

At present, I have completed the development of the Drawer component locally.

![Screen Recording 2024-01-02 at 00 15 42](https://github.com/nextui-org/nextui/assets/31227919/02144aa2-2a0b-4fa8-a6e1-4a831f6b314d)
